### PR TITLE
azure-podvm-image-build: Make image name unique

### PR DESCRIPTION
**Pull request by surajssd**
Fixes: #916

### CODE REVIEW

**Comment by surajssd** [comment]
Testing done, follow the steps until the step ["Build Pod VM Image" in this document](https://github.com/confidential-containers/cloud-api-adaptor/blob/staging/azure/README.md#build-pod-vm-image): 

Output of 1st run:

```bash
$ PODVM_DISTRO=ubuntu make image && cd -
cd "../.." && ARCH=amd64 make agent-protocol-forwarder
...
==> peer-pod-ubuntu.azure-arm.ubuntu:
==> peer-pod-ubuntu.azure-arm.ubuntu: The resource group was not created by Packer, not deleting ...
Build 'peer-pod-ubuntu.azure-arm.ubuntu' finished after 10 minutes 19 seconds.

==> Wait completed after 10 minutes 19 seconds

==> Builds finished. The artifacts of successful builds are:
--> peer-pod-ubuntu.azure-arm.ubuntu: Azure.ResourceManagement.VMImage:

OSType: Linux
ManagedImageResourceGroupName: suraj-caa-az-cvm4
ManagedImageName: peer-pod-vmimage-0.0.1
ManagedImageId: /subscriptions/.../resourceGroups/suraj-caa-az-cvm4/providers/Microsoft.Compute/images/peer-pod-vmimage-0.0.1
ManagedImageLocation: eastus
ManagedImageSharedImageGalleryId: /subscriptions/.../resourceGroups/suraj-caa-az-cvm4/providers/Microsoft.Compute/galleries/caaubntcvmsGallery/images/cc-image/versions/0.0.1
SharedImageGalleryResourceGroup: suraj-caa-az-cvm4
SharedImageGalleryName: caaubntcvmsGallery
SharedImageGalleryImageName: cc-image
SharedImageGalleryImageVersion: 0.0.1
SharedImageGalleryReplicatedRegions: eastus
```

Output of 2nd run:

```bash
$ export PKR_VAR_az_gallery_image_version=0.0.2
$ PODVM_DISTRO=ubuntu make image && cd -
cd "../.." && ARCH=amd64 make agent-protocol-forwarder
...
Build 'peer-pod-ubuntu.azure-arm.ubuntu' finished after 11 minutes 59 seconds.

==> Wait completed after 11 minutes 59 seconds

==> Builds finished. The artifacts of successful builds are:
--> peer-pod-ubuntu.azure-arm.ubuntu: Azure.ResourceManagement.VMImage:

OSType: Linux
ManagedImageResourceGroupName: suraj-caa-az-cvm4
ManagedImageName: peer-pod-vmimage-0.0.2
ManagedImageId: /subscriptions/.../resourceGroups/suraj-caa-az-cvm4/providers/Microsoft.Compute/images/peer-pod-vmimage-0.0.2
ManagedImageLocation: eastus
ManagedImageSharedImageGalleryId: /subscriptions/.../resourceGroups/suraj-caa-az-cvm4/providers/Microsoft.Compute/galleries/caaubntcvmsGallery/images/cc-image/versions/0.0.2
SharedImageGalleryResourceGroup: suraj-caa-az-cvm4
SharedImageGalleryName: caaubntcvmsGallery
SharedImageGalleryImageName: cc-image
SharedImageGalleryImageVersion: 0.0.2
SharedImageGalleryReplicatedRegions: eastus
```

File: azure/image/ubuntu/azure-ubuntu.pkr.hcl [hunk 1]
```diff
@@ -9,7 +9,7 @@ source "azure-arm" "ubuntu" {
   image_publisher                   = "${var.publisher}"
   image_offer                       = "${var.offer}"
   image_sku                         = "${var.sku}"
-  managed_image_name                = "${var.az_image_name}"
+  managed_image_name                = "${var.az_image_name}-${var.az_gallery_image_version}"
```
Comment (bpradipt) [inline azure/image/ubuntu/azure-ubuntu.pkr.hcl:12]: Why not override the `az_image_name` during start of the build? This way it leaves it to the user to decide how to name the image.

Comment (surajssd) [reply azure/image/ubuntu/azure-ubuntu.pkr.hcl:12]: Yes I agree. A user can choose to provide an image name with that format if they like. `PKR_VAR_az_image_name="$IMAGE_NAME-$PKR_VAR_az_gallery_image_version"`

**Comment by surajssd** [comment]
Closing this, leaving the choice for the user to decide what they would want.

**Pull request closed without merging**